src/app_callback_service.rs

```diff
@@ -0,0 +1,69 @@
+use tonic::{Request, Response, Status};
+
+use dapr::{
+    appcallback::*,
+    dapr::dapr::proto::runtime::v1::app_callback_server::AppCallback,
+};
+
+#[derive(Default)]
+pub struct AppCallbackService {}
+
+#[tonic::async_trait]
+impl AppCallback for AppCallbackService {
+    /// Invokes service method with InvokeRequest.
+    async fn on_invoke(
+        &self,
+        _request: Request<InvokeRequest>,
+    ) -> Result<Response<InvokeResponse>, Status> {
+        Ok(Response::new(InvokeResponse::default()))
+    }
+
+    /// Lists all topics subscribed by this app.
+    ///
+    /// NOTE: Dapr runtime will call this method to get
+    /// the list of topics the app wants to subscribe to.
+    /// In this example, the app is subscribing to topic `A`.
+    async fn list_topic_subscriptions(
+        &self,
+        _request: Request<()>,
+    ) -> Result<Response<ListTopicSubscriptionsResponse>, Status> {
+        let topic = "product-variant/created".to_string();
```

Comment:
I think the correct event would start with `catalog/`
otherwise looks good
I will also document the events in the wiki soon